src/platform/plugins/shared/content_management/server/routes/list/route.ts

```diff
@@ -0,0 +1,377 @@
+          const hasCreatedByFilter = createdBy && createdBy.length > 0;
+
+          if (hasCreatedByFilter) {
+            const hasNonUidValues = createdBy.some((v) => !isUserProfileUid(v));
+
+            if (hasNonUidValues) {
+              // Fetch distinct creators from the index to resolve usernames/emails.
+              const distinctCreators = await getDistinctCreators(
+                savedObjectsClient,
+                type,
+                namespaces,
+                logger
+              );
+              const resolveResult = await resolveCreatedByFilter(
+                createdBy,
+                distinctCreators,
+                coreStart,
+                logger
+              );
+              resolvedCreatedBy = resolveResult.uids;
+              createdByInputToUidMap = resolveResult.inputToUidMap;
+
+              // If user requested createdBy filter but no UIDs resolved, use a placeholder
+              // that will never match to ensure zero results are returned.
+              if (resolvedCreatedBy.length === 0) {
+                resolvedCreatedBy = ['__no_match__'];
+              }
+            } else {
+              // All values are UIDs, no resolution needed.
+              resolvedCreatedBy = createdBy;
+              // Build identity mapping for UIDs.
+              createdByInputToUidMap = {};
+              for (const uid of createdBy) {
+                createdByInputToUidMap[uid] = uid;
+              }
+            }
+          }
+
+          // Build the ES query.
+          const query = buildSearchQuery({
+            searchQuery,
+            tags,
+            favoritesOnly,
+            favoriteRawIds,
+            createdBy: resolvedCreatedBy,
+            type,
+          });
+
+          // Perform the search.
+          const result = await savedObjectsClient.search({
+            type,
+            namespaces,
+            query,
+            runtime_mappings: buildRuntimeMappings(sort.field),
+            sort: buildSort(sort.field, sort.direction),
+            from: page.index * page.size,
+            size: page.size,
+            track_total_hits: true,
+          });
+
+          // Transform the response.
+          const transformedItems = transformHits(result.hits.hits, additionalAttributes);
+          const total =
+            typeof result.hits.total === 'number'
+              ? result.hits.total
+              : result.hits.total?.value ?? 0;
+
+          // Collect unique user IDs for profile enrichment.
+          const userIds = new Set<string>();
+          for (const item of transformedItems) {
+            if (item.createdBy) userIds.add(item.createdBy);
+            if (item.updatedBy) userIds.add(item.updatedBy);
+          }
+
+          // Fetch user profiles and enrich items.
+          const userInfoMap = await fetchUserProfiles(Array.from(userIds), coreStart, logger);
```

Comment:
if we fetch 100 dashboards and there is 1 creator - we return the same user profile 100x 

Maybe we should keep it as today where it is enough to return profile_id and client-side we have smart components with react query that know how to render profiles from user cache <UserAvatar id={id} />  ?

or at least return profiles as separate map, but I'd prefer not include them at all

Reply:
`tag:(clint.hall@elastic.co)` 

I think this is a cool feature, but since it's new, we might need more consideration before merging it in production. There are definitely some edge cases we should think through. 

bulkGet might be a bottleneck when fetching 50+ users: 

https://github.com/elastic/kibana/blob/8773c7482f5ba5b245724ff351cc5ab83b558dd0/x-pack/platform/plugins/shared/security/server/user_profile/user_profile_service.ts#L257-L281